deck.gl__core/index.d.ts

```diff
@@ -2409,23 +2410,23 @@ declare module "@deck.gl/core/lib/deck" {
 			};
 			oldViewState: any;
 		}) => any;
-		onHover?: <D>(info: PickInfo<D>, e: MouseEvent) => any;
-		onClick?: <D>(info: PickInfo<D>, e: MouseEvent) => any;
-		onDragStart?: <D>(info: PickInfo<D>, e: MouseEvent) => any;
-		onDrag?: <D>(info: PickInfo<D>, e: MouseEvent) => any;
-		onDragEnd?: <D>(info: PickInfo<D>, e: MouseEvent) => any;
-		onLoad?: () => void;
-		onResize?: (size: { height: number; width: number }) => void;
-		onBeforeRender?: (args: { gl: WebGLRenderingContext }) => void;
-		onAfterRender?: (args: { gl: WebGLRenderingContext }) => void;
-		onError?: (error: Error, source: any) => void;
-		_onMetrics?: (metrics: MetricsPayload) => void;
+		onHover: <D>(info: PickInfo<D>, e: MouseEvent) => any;
+		onClick: <D>(info: PickInfo<D>, e: MouseEvent) => any;
+		onDragStart: <D>(info: PickInfo<D>, e: MouseEvent) => any;
+		onDrag: <D>(info: PickInfo<D>, e: MouseEvent) => any;
+		onDragEnd: <D>(info: PickInfo<D>, e: MouseEvent) => any;
+		onLoad: () => void;
+		onResize: (size: { height: number; width: number }) => void;
+		onBeforeRender: (args: { gl: WebGLRenderingContext }) => void;
+		onAfterRender: (args: { gl: WebGLRenderingContext }) => void;
+		onError: (error: Error, source: any) => void;
+		_onMetrics: (metrics: MetricsPayload) => void;
 
-		ContextProvider?: React.Provider<ContextProviderValue>
+		ContextProvider: React.Provider<ContextProviderValue>
 	}
 
 	export default class Deck {
-		constructor(props: DeckProps);
+		constructor(props: Partial<DeckProps>);
```

Comment:
Do we need to make `props` a `Partial` as well?

Reply:
I think so, otherwise you have to provide values for all properties.

Reply:
Sorry, I meant the field `props` a few lines down in the `Deck` class, not the parameter here.

Reply:
As far as I can see are all parameter either default or overridden in the constructor. So, they should all be set and `Partial<>` isn't needed.

Reply:
Thanks for looking into that!